src/hiopbbpy/opt/optproblem.py

```diff
@@ -16,38 +16,48 @@
 Authors:    Tucker Hartland <hartland1@llnl.gov>
             Nai-Yuan Chiang <chiang7@llnl.gov>
 """
+from typing import Callable, Dict, List, Union, Tuple
+
 class IpoptProb:
-    def __init__(self, objective, gradient, constraints_list, xbounds, solver_options=None):
-        self.constraints_list = constraints_list
+    def __init__(self, objective, gradient, constraint:Union[Dict, List[Dict]], xbounds, solver_options=None):
+        self.cons = constraint
         self.eval_f = objective
         self.eval_g  = gradient
         self.xl = [b[0] for b in xbounds]
         self.xu = [b[1] for b in xbounds]
         self.cl = []
         self.cu = []
         self.nvar = len(xbounds)
-        self.ncon = len(self.constraints_list)
+
         self.ipopt_options = solver_options
 
-        for con in constraints_list:
-            if con['type'] == 'eq':
-                self.cl.append(0.0)
-                self.cu.append(0.0)
-            elif con['type'] == 'ineq':
-                self.cl.append(0.0)
-                self.cu.append(np.inf)
-            else:
-                raise ValueError(f"Unknown constraint type: {con['type']}")
+        if isinstance(self.cons, list):
+            # constraints is provided as a list of dict, supported by SLSQP and Ipopt
+            for con in self.cons:
+                if con['type'] == 'eq':
```

Comment:
What if the `con` dictionary does not have key `type`?

Reply:
I added a function to check it. Note that this is also checked by SLSQP.

Reply:
Great!